src/smk/tool/geomark/tool-geomark.js

```diff
@@ -140,6 +140,18 @@ include.module( 'tool-geomark', [
                 currentDrawingLayer.addLayer(eventLayer);
             }
 
+            this.drawStart = function(e1) {
+                e1.workingLayer.on('pm:vertexadded', function(e2) {
```

Comment:
Since you're doing some cleanup, we should unsubscribe from the pm:vertexadded event as well.

Reply:
We already unsubscribe this function from the 'pm:drawstart' event, so my thinking (and reading https://leafletjs.com/reference.html#map-off) is that this would be sufficient, because now map is no longer listening for this event with this function. Also, I'm not sure how we'd unsubscribe 'pm:vertexadded' events and function because we don't have a reference on the layers at this point. Let me know if I'm missing something.

Reply:
I also forgot to mention this earlier ... I also tested with a breakpoint in the function that we subscribe 'pm:vertexadded' on, and confirmed that when I leave the Geomark panel and draw shapes using Markup, the breakpoint is not called.

Reply:
Any time you subscribe to an event you really should be unsubscribing as well, or else you can end up with memory leaks. Probably not a huge issue for SMK and the geomark tool, but still a valid pattern to follow. Here's how you accomplish this:
1. create a new variable (eg. var activeDrawingLayer) to hold a reference to the 'working layer'
2. Extract the pm:vertexadded event handler (ie. the inline anonymous function(e2)) to a named function (eg. this.setCanClear())
3. Set this new function as the event handler for the pm:vertexadded event. (eg e1.workingLayer.on('pm:vertexadded', self.setCanClear))
4. In this.drawStart assign e1.workingLayer to your new variable so you have a reference to the layer that the drawing tool is using
5. In this.drawEnd unsubscribe (eg. activeDrawingLayer.off('pm:vertexadded',  self.setCanClear)

Reply:
Thanks for this suggestion! I've implemented it and it tests well.

Because "activeDrawingLayer" is similar to the existing variable "currentDrawingLayer", I'm renaming the latter as "currentLayerGroup".

Reply:
Looks great!